mender-install-files.sh

```diff
@@ -0,0 +1,311 @@
+    [[ ${#mappings[@]} -eq 0 ]] && \
+        { echo "Error: partition mappings failed. Aborting."; exit 1; } || \
+        { echo "Mapped ${#mappings[@]} partition(s)."; }
+  else
+    echo "Error: no device passed. Aborting."
+    exit 1
+  fi
+
+  echo $sudoPW | sudo -S partprobe /dev/${mappings[0]%p*}
+}
+
+detach_mappings() {
+  for mapping in ${mappings[@]}
+  do
+    map_dev=/dev/mapper/"$mapping"
+    is_mounted=`grep ${map_dev} /proc/self/mounts | wc -l`
+    if [ ${is_mounted} -ne 0 ]; then
+      echo "Unmounting detected mounted mapping: $mapping"
+      echo $sudoPW | sudo -S umount -l $map_dev
+    fi
+  done
+
+  mapper=${mappings[0]%p*}
+  echo "Detach mappings: /dev/$mapper"
+  echo $sudoPW | sudo -S kpartx -d /dev/$mapper &
+  echo $sudoPW | sudo -S losetup -d /dev/$mapper &
+  wait
+}
+
+create_client_files() {
+  cat <<- EOF > $BASE_DIR/mender.service
+	[Unit]
+	Description=Mender OTA update service
+	After=systemd-resolved.service
+
+	[Service]
+	Type=idle
+	User=root
+	Group=root
+	ExecStartPre=/bin/mkdir -p -m 0700 /data/mender
+	ExecStartPre=/bin/ln -sf /etc/mender/tenant.conf /var/lib/mender/authtentoken
+	ExecStart=/usr/bin/mender -daemon
+	Restart=on-abort
+
+	[Install]
+	WantedBy=multi-user.target
+	EOF
+  cat <<- EOF > $BASE_DIR/mender.conf
+	{
+	    "InventoryPollIntervalSeconds": 5,
+	    "RetryPollIntervalSeconds": 1,
+	    "RootfsPartA": "/dev/mmcblk0p2",
+	    "RootfsPartB": "/dev/mmcblk0p3",
+	    "ServerCertificate": "/etc/mender/server.crt",
+	    "ServerURL": "https://docker.mender.io",
+	    "TenantToken": "dummy",
+	    "UpdatePollIntervalSeconds": 5
+	}
+	EOF
+  # mender-image-1.4.0
+  cat <<- EOF > $BASE_DIR/artifact_info
+	artifact_name=${artifact}
+	EOF
+  # Version file
+  echo -n "2" > $BASE_DIR/version
+  # beaglebone
+  cat <<- EOF > $BASE_DIR/device_type
+	device_type=${type}
+	EOF
+  cat <<- EOF > $BASE_DIR/fw_env.config
+	/dev/mmcblk0 0x800000 0x20000
+	/dev/mmcblk0 0x1000000 0x20000
+	EOF
+}
+
+update_etc_hosts() {
+  local primary=$1
+  echo "$server docker.mender.io s3.docker.mender.io" | sudo tee -a $primary/etc/hosts
+}
+
+get_mender_repos() {
+  mkdir -p $GOPATH
+  export GOPATH
+  
+  [ -z $gobin ] && gobin="/usr/local/go/bin"
+  $gobin/go get ${GO_IMPORT}
+
+  git clone https://github.com/mendersoftware/meta-mender.git
+
+  cp ${BASE_DIR}/meta-mender/meta-mender-core/recipes-bsp/grub/files/fw_printenv ${BASE_DIR}/
+
+  rm -rf ${BASE_DIR}/meta-mender
+}
+
+install_files() {
+  local primary=$1
+  local data=$2
+
+  identitydir="usr/share/mender/identity"
+  inventorydir="usr/share/mender/inventory"
+  sysconfdir="etc/mender"
+  bindir="usr/bin"
+  systemd_unitdir="lib/systemd/system"
+  localstatedir="var/lib/mender"
+  base_sbindir="sbin"
+  dataconfdir="mender"
+  databootdir="u-boot"
+
+  # Prepare 'data' partition
+  echo $sudoPW | sudo -S install -d -m 755 ${data}/${dataconfdir}
+  echo $sudoPW | sudo -S install -d -m 755 ${data}/${databootdir}
```

Comment:
The `install` commands should not to require `sudo`. This applies to the all the locations of `install`. Why the use of sudo?

If `sudo` is required after all, I would recommend do all the install steps on a "staging area" and do an rsync/copy of staging area to actual rootfs with `sudo`, to reduce number of `sudo` calls.

Reply:
OK.